cds_dojson/marc21/fields/books/utils.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of CERN Document Server.
+# Copyright (C) 2017, 2018, 2019 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+"""Book utils."""
+
+import re
+
+
+MAX_PAGES_NUMBER = 8192
+
+
+def is_excluded(value):
+    """Validate if field 300 should be excluded."""
+    exclude = [
+        "mul. p",
+        "mult p",
+        "mult. p",
+        "mult. p.",
+        "multi p",
+        "multi pages",
+    ]
+    if not value or value.lower() in exclude:
+        return True
+    return False
+
+
+def extract_number_of_pages(value):
+    """Extract number of pages from 300 if exists."""
+    res = re.findall(r"([0-9]+) *p", value, flags=re.IGNORECASE)
+
+    # If we have more than one occurency of pages its UnexpectedValue
+    if len(res) == 1 and int(res[0]) < MAX_PAGES_NUMBER:
+        return int(res[0])
+    return None
+
+
+def extract_physical_description(value):
+    """Extract physical description from 300 if any."""
+    res = re.findall(
+        r"([0-9]+ \w[CD\-ROM|DVD\-ROM|diskette|VHS]+)",
```

Comment:
Did you notice if there was also `CDROM` or `DVDROM` so you can have the `-` optional?

Reply:
Yes there are very few (less than five cases), I forced the hyphen for the shake of consistency, so we don't end up with all possible combinations of CD-ROM(s).